include/Core/Posts/Metabox/DeletePostsByCustomTaxonomMetabox.php

```diff
@@ -0,0 +1,195 @@
+		}
+
+		if ( count( $terms_array ) > 0 ) {
+?>
+        <!-- Custom tax Start-->
+        <h4><?php _e( 'Select the post type from which you want to delete posts by custom taxonomy', 'bulk-delete' ); ?></h4>
+
+        <fieldset class="options">
+            <table class="optiontable">
+				<?php bd_render_post_type_dropdown( 'tax' ); ?>
+            </table>
+
+            <h4><?php _e( 'Select the taxonomies from which you want to delete posts', 'bulk-delete' ) ?></h4>
+
+            <table class="optiontable">
+<?php
+			foreach ( $terms_array as $tax => $terms ) {
+?>
+                <tr>
+                    <td scope="row" >
+                        <input name="smbd_taxs" value="<?php echo $tax; ?>" type="radio" class="custom-tax">
+                    </td>
+                    <td>
+                        <label for="smbd_taxs"><?php echo $taxs[$tax]->labels->name; ?> </label>
+                    </td>
+                </tr>
+<?php
+			}
+?>
+            </table>
+
+            <h4><?php _e( 'The selected taxonomy has the following terms. Select the terms from which you want to delete posts', 'bulk-delete' ) ?></h4>
+            <p><?php _e( 'Note: The post count below for each term is the total number of posts in that term, irrespective of post type', 'bulk-delete' ); ?>.</p>
+<?php
+			foreach ( $terms_array as $tax => $terms ) {
+?>
+            <table class="optiontable terms_<?php echo $tax;?> terms">
+<?php
+				foreach ( $terms as $term ) {
+?>
+                    <tr>
+                        <td scope="row" >
+                            <input name="smbd_tax_terms[]" value="<?php echo $term->slug; ?>" type="checkbox" class="terms">
+                        </td>
+                        <td>
+                            <label for="smbd_tax_terms"><?php echo $term->name; ?> (<?php echo $term->count . ' '; _e( 'Posts', 'bulk-delete' ); ?>)</label>
+                        </td>
+                    </tr>
+<?php
+				}
+?>
+            </table>
+<?php
+			}
+?>
+            <table class="optiontable">
+				<?php
+				$this->render_filtering_table_header();
+				$this->render_restrict_settings();
+				$this->render_delete_settings();
+				$this->render_limit_settings();
+				$this->render_cron_settings();
+				?>
+            </table>
+
+            </fieldset>
+<?php
+			$this->render_submit_button();
+		} else {
+?>
+            <h4><?php _e( "This WordPress installation doesn't have any non-empty custom taxonomies defined", 'bulk-delete' ) ?></h4>
+<?php
+		}
+	}
+
+	protected function convert_user_input_to_options( $request, $options ) {
+		$options                       = array();
+		$options['post_type']          = bd_array_get( $request, 'smbd_tax_post_type', 'post' );
+		$options['selected_taxs']      = bd_array_get( $request, 'smbd_taxs' );
+		$options['selected_tax_terms'] = bd_array_get( $request, 'smbd_tax_terms' );
+		$options['restrict']           = bd_array_get_bool( $request, 'smbd_taxs_restrict', false );
```

Comment:
Apart from `post_type`, `selected_taxs` and `selected_tax_terms`, you can remove the others.

They are handled in the `process_common_filters` method.

Reply:
@sudar this code have some cnflicts, I could not merge it using my git GUI enveronent. I hope it ll be fixed after merging